src/utils/upload/sampleInspector.js

```diff
@@ -0,0 +1,136 @@
+import {
+  DecodeUTF8, Decompress,
+} from 'fflate';
+
+const Verdict = {
+  INVALID_BARCODES_FILE: 'INVALID_SAMPLE_FILES',
+  INVALID_FEATURES_FILE: 'INVALID_FEATURES_FILE',
+  INVALID_TRANSPOSED_MATRIX: 'INVALID_TRANSPOSED_MATRIX',
+};
+
+const verdictText = {
+  [Verdict.INVALID_BARCODES_FILE]: 'Barcodes file is invalid',
+  [Verdict.INVALID_FEATURES_FILE]: 'Features file is invalid',
+  [Verdict.INVALID_TRANSPOSED_MATRIX]: 'Matrix file is transposed',
+};
+
+const CHUNK_SIZE = 2 ** 18; // 250 kb
```

Comment:
What is this chunk size?

Reply:
chunk size is the size of file that we want to process at a time. To process the file, we don't open the file at once, but take it part by part (as large as chunk sizes). This allows us to process large files without having to load all of them into memory

Reply:
that's great! could you add a comment explaining that?

Reply:
Turns out we can't arbitrarily take chunks out of files and read them separately. Part of the GZIP compression works by changing repetitive characters into "pointers" to a previous part located in the earlier part of the file. If we take a middle slice of the file arbitrarily, some pointers will lose reference to the earlier part because they are not included in the slice.

The file is divided into blocks which, I think, should be able to be decoded independently. However, this will require more reading time and understanding, because it seems that block sizes might vary. More info on GZIP compression: https://commandlinefanatic.com/cgi-bin/showarticle.cgi?article=art001

Looking at available samples, the bulk of data is located in the matrix file. The size of features and barcodes files are not much (in the order of 10 MB). For these files, I think we can decompress them into memory without slicing them.